### PR TITLE
add String() method to User

### PR DESCRIPTION
**Pull request by ammario**

### CODE REVIEW

**Comment by iopred** [comment]
It might be good to move the User struct into this file, similar to Message.go

File: user_test.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package discordgo
+
+import "testing"
+
+func TestUser(t *testing.T) {
+	t.Parallel()
+
+	t.Run("String()", func(t *testing.T) {
```
Comment (iopred) [inline user_test.go]: 1.6 doesn't support Run so Travis is breaking, could you just inline the inside of this method to TestUser? We don't **need** this test to run in parallel :)

File: user.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	Discriminator string `json:"discriminator"`
+	Token         string `json:"token"`
+	Verified      bool   `json:"verified"`
+	MFAEnabled    bool   `json:"mfa_enabled"`
+	Bot           bool   `json:"bot"`
+}
+
+//String returns a unique identifier of the form username#discriminator
+func (u *User) String() string {
+	return fmt.Sprintf("%v#%v", u.Username, u.Discriminator)
```
Comment (iopred) [inline user.go]: Sorry, last thing, %s is the correct way to format a string. `"%s#%s"` would work best here

File: user_test.go (hunk 1 above)
Comment (ammario) [inline user_test.go]: Sure, done w/ e19d4f4acb53dd048c20bd9676e1f823a6b3b9d6. The t.Parallel call should work with 1.6, I used the `t.Run` for organization not parallelism

**Comment by iopred** [comment]
Thanks for the PR! <3

**Pull request merged**